frida_tools/application.py

```diff
@@ -445,6 +451,47 @@ def work():
 
         return result[0]
 
+    def return_real_args(self, parser):
+        real_args = sys.argv[1:]
+        performPass = True
+        files_processed = set()
+
+        while (performPass):
+            offset = self.find_arg_file_offset(real_args, parser)
+            if (offset < 0):
+                performPass = False
+                continue
+
+            file_path = real_args[offset+1]
+            
+            if file_path in files_processed:
+                parser.error('File "' + file_path + '" given twice as -O argument')
+
+            if os.path.isfile(file_path):
+                with codecs.open(file_path, 'r', 'utf-8') as f:
+                  new_arg_text = f.read()
+            else:
+                parser.error('File "' + file_path + '" following -O option is not a valid file')
+
+            real_args = self.insert_new_args_in_list(real_args, offset, new_arg_text)
+            files_processed.add(file_path)
+
+        return real_args
+
+    def find_arg_file_offset(self, arglist, parser):
+        for i in range(len(arglist)):
+            if (arglist[i] == '-O') or (arglist[i] == '--options-file'):
+                if i < len(arglist)-1:
+                    return i
+                else:
+                    # No argument following the option, error
+                    parser.error ('No argument given for -O option')
+        return -1
+
+    def insert_new_args_in_list(self, args, offset, new_arg_text):
+        new_args = shlex.split(new_arg_text)
+        new_args_list = args[:offset] + new_args + args[offset+2:]
```

Comment:
```suggestion
        new_args_list = args[:offset] + new_args + args[offset + 2:]
```

Reply:
Done.  What coding rule is it to pad the '+' operator with spaces?

Reply:
It's defined in [PEP-8](https://www.python.org/dev/peps/pep-0008/), which our style is based on. (Our only addition is the double quotes vs single quotes rule based on string vs enum.)